daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1251,6 +1251,28 @@ private[lf] object SBuiltin {
     }
   }
 
+  // Convert an interface `requiredIface` to another interface `requiringIface`, if
+  // the `requiringIface` implements `requiredIface`.
+  final case class SBFromRequiredInterface(
+      requiringIface: TypeConName
+  ) extends SBuiltin(1) {
+
+    override private[speedy] def execute(
+        args: util.ArrayList[SValue],
+        machine: Machine,
+    ) = {
+      val record = getSRecord(args, 0)
+      // TODO https://github.com/digital-asset/daml/issues/11345
+      //  The lookup is probably slow. We may want to investigate way to make the feature faster.
+      machine.returnValue = machine.compiledPackages.interface.lookupTemplate(record.id) match {
+        case Right(ifaceSignature) if ifaceSignature.implements.contains(requiringIface) =>
```

Comment:
As pointed offline by @sofiafaro-da, this is broken. 
We should not check that `ifaceSignature` implements `requiringIface`, but requires (transitively) `requiringIface`, but not `implements`. The later implies the former, but the reverse is not true.